src/lightning_app/storage/drive.py

```diff
@@ -33,6 +35,8 @@ def __init__(
             component_name: The component name which owns this drive.
                 When not provided, it is automatically inferred by Lightning.
             root_folder: This is the folder from where the Drive perceives the data (e.g this acts as a mount dir).
+            optimization: Whether to optimize the drive with a given option. This also defines if the drive will be
+                mounted to the component and accessible via normal file system operations.
```

Comment:
Can the documentation here list what the possible values are? Or even better, have it part of the type annotation if possible. 

Another question I have is: Doesn't this leak too much infrastructure knowledge into the framework? As a user, why would I ever have optimization turned off? Can't we just have it enabled by default? If it's not just about turning thing on or off, how do I know which option to choose and when?

The docs should probably also mention that the value set here gets ignored when running locally. 
Not saying it should all be explained here in the api. Most of this can probably be clarified in user docs.

Reply:
It's tricky, but generally you can have multiple optimizations for a given protocol. This is not something that will happen immediately, but this is the general direction in which we're going. 

For example for `s3://` you could have `indexed-s3` (Fast, read-only, no cost), `direct-mount` (Slower, read-write, no cost) or `fsx` (Fastest, read-write, expensive) and these would all be expressed via protocol + optimization combinations. Similarly it is possible that the `indexed-s3` optimization will be made available for `lit://` drives to mount them to a work as a file system.

Reply:
@panos-is I think it would be better to abstract those within a table.

A user won't understand the difference between `indexed-s3`, etc. could be replaced 'default', 'fast', 'fastest'.

Reply:
@tchaton I don't really agree. We need to expose some of these ideas here otherwise it becomes very hard to figure out what these words really mean. We'll still have a default for each protocol, so it should be easy to utilize the base functionality without engaging with this at all.

Reply:
This would be documented within a table in the documentation and in the docstring. But I don't know what indexed-s3 means as a user. This doesn't provide value to me. But I understand optimization="fatest"

Reply:
That makes sense to me too. For example, we have something similar for our compute where abstract away the type of compute instance.

https://lightning.ai/lightning-docs/core_api/lightning_work/compute.html?highlight=compute

But ideally, we should have a lookup table within our documentation explaining what "fastest" would mean.

Reply:
I don't think this works like that, since different opts will have very different properties besides just speed (read-writedness, full or only forward sync etc) but for the time being we can just remove this and add a mount boolean flag and not have to resolve this discussion to unblock this.

Reply:
After conversing with @panos-is I devided to remove this. For now the only thing we need to know is that if an `s3://` drive is used, that will be automatically mounted. the optimization type is something we will work on later.